myapp/logic/trade_logic.py

```diff
@@ -0,0 +1,34 @@
+from myapp import stock_api
+from myapp import models
+
+def extract_info(params):
+    try:
+        number_of_stocks = int(params["number_of_stocks"])
+        if number_of_stocks <= 0:
+            raise Exception
```

Comment:
we discucssed this last week. If you are going to use Exceptions to indicate failure you should use specific, custom exceptions. ie NotEnoughMoneyException. These can be caught outside the function and handled appropriately.

Reply:
Yes, Done!